src/main/flight/gps_rescue.c

```diff
@@ -51,15 +51,74 @@
 
 #include "gps_rescue.h"
 
+typedef enum {
+    RESCUE_SANITY_OFF = 0,
+    RESCUE_SANITY_ON,
+    RESCUE_SANITY_FS_ONLY
+} gpsRescueSanity_e;
+
+typedef enum {
+    RESCUE_IDLE,
+    RESCUE_INITIALIZE,
+    RESCUE_ATTAIN_ALT,
+    RESCUE_CROSSTRACK,
+    RESCUE_LANDING_APPROACH,
+    RESCUE_LANDING,
+    RESCUE_ABORT,
+    RESCUE_COMPLETE
+} rescuePhase_e;
+
+typedef enum {
+    RESCUE_HEALTHY,
+    RESCUE_FLYAWAY,
+    RESCUE_CRASH_DETECTED,
+    RESCUE_TOO_CLOSE
+} rescueFailureState_e;
+
+typedef struct {
+    int32_t targetAltitudeCm;
+    int32_t targetGroundspeed;
+    uint8_t minAngleDeg;
+    uint8_t maxAngleDeg;
+    bool crosstrack;
+} rescueIntent_s;
+
+typedef struct {
+    int32_t maxAltitudeCm;
+    int32_t currentAltitudeCm;
+    uint16_t distanceToHomeM;
+    uint16_t maxDistanceToHomeM;
+    int16_t directionToHome;
+    uint16_t groundSpeed;
+    uint8_t numSat;
+    float zVelocity; // Up/down movement in cm/s
+    float zVelocityAvg; // Up/down average in cm/s
+    float accMagnitude;
+    float accMagnitudeAvg;
+} rescueSensorData_s;
+
+typedef struct {
+    bool bumpDetection;
+    bool convergenceDetection;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [struct member 'rescueSanityFlags::convergenceDetection' is never used.](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5552604)